### PR TITLE
Hotfix/handle numeric string

### PR DESCRIPTION
**Pull request by lapistano**
this patch makes it possible to use per example:

```
?q=eq(size,"12") 
?q=eq(size,'12') 
```

as rql query string.

### CODE REVIEW

**Comment by hairmare** [comment]
If I understand this correctly this would make `12 == '12'` and so on. I'm not convinced that this should be the behaviour we support. While I did do a quick test using '' on a numeric field, I'm actually not at all sure what should happen in that case. I feel doing to much autocasting ala php won't be of much help.

We should also stop writing any more `LogicException` tests before finishing #17. Personally I think that should be a more pressing matter compared to what you are fixing here.

File: test/LexerTest.php [hunk 1]
```diff
@@ -188,4 +238,47 @@ public function concatenatorProvider()
...
+     * test set of field concatenators
+     *
+     * @dataProvider quotationCharProvider
+     *
+     * @param boolean $tmp           not used
+     * @param string  $quotationChar Character representing a string quotationChar.
+     *
+     * @return void
+     */
+    public function testIsFieldQuotationChar($tmp, $quotationChar)
```
Comment (hairmare) [inline test/LexerTest.php]: What the `$tmp`?

**Comment by lapistano** [comment]
You got it wrong. 
This will enable the possibility to indicate a number as a string by adding single or double quotes, but keep the functionality to recognize numbers as integers for the search

File: test/LexerTest.php (hunk 1 above)
Comment (lapistano) [inline test/LexerTest.php]: this is a dummy param, so I could reuse the dataProvider... but yeah, I should refactor that ..

File: test/LexerTest.php [hunk 2]
```diff
@@ -39,126 +39,176 @@ public function testLexer($rql, $expected)
     public function lexerProvider()
     {
         return array(
-            'complex eq search concatenated fields' => array('eq(name-part+test,12)', array(
```
Comment (hairmare) [inline test/LexerTest.php:42]: Please revert you syntax cleanup on this file. I'll be glad to merge the cleanup in another PR. Having it here makes it impossible to review this PR.

**Comment by lapistano** [comment]
As for the tests on exceptions .. they are easily removed/refactored once #17 is implemented .. but I really want to see the exceptions tested.

File: test/LexerTest.php (hunk 2 above)
Comment (lapistano) [inline test/LexerTest.php:42]: I can see what I can do .. PHPStorm does this on save ;)

File: test/ParserTest.php [hunk 3]
```diff
@@ -41,6 +41,16 @@ public function parserProvider()
 
         $eqAST = new AST\EqOperation;
         $eqAST->setProperty('name');
+        $eqAST->setValue('"12"');
+        $tests['simple eq with numeric in double quotes'] = array('eq(name,"12")', $eqAST);
```
Comment (hairmare) [inline test/ParserTest.php:46]: I'm getting something wrong but am not sure what. Isn't this now `name === '12'` where name must be a string? Would `eq(size,'12')` result in `size === '12'` (ie. always false) when size is an int? Looking at other implementations it seems that rql does implicit strings all the time.

What about `eq(name,avec'apos)` and `eq(name,"Hans 'Housi' Wale-Sepp")`?

File: test/LexerTest.php (hunk 2 above)
Comment (hairmare) [inline test/LexerTest.php:42]: As said, I'll gladly merge it in another PR...

Comment (lapistano) [inline test/LexerTest.php:42]: Patience, my padavan! On it I am.

Comment (hairmare) [inline test/LexerTest.php:42]: Patience I have worry you not. The IDE wars ignite again I shall not o_O.

File: test/ParserTest.php (hunk 3 above)
Comment (lapistano) [inline test/ParserTest.php:46]: Will add both to the testcases .. will see, if they break s.th. thx for the hint.

**Comment by hairmare** [comment]
The thing whith `LogicException` is that it is the wrong exception in all the cases it's being used. I didn't test them since this needs to get changed before we hit `2.0.0` anyway and I didn't want to document any behaviour that we need to obsolete ASAP.

File: test/LexerTest.php (hunk 2 above)
Comment (lapistano) [inline test/LexerTest.php:42]: Blaming my IDE too early I did. The testcases to the beginning of the array I added

**Comment by lapistano** [comment]
It sure is the wrong exception ... I only stuck to the pattern to be able to find them (also newly added) in one go ;)

**Comment by lapistano** [comment]
this would be mergable, if @hairmare does not mind.

File: test/LexerTest.php (hunk 2 above)
Comment (hairmare) [inline test/LexerTest.php:42]: It's still a case of this:

``` php
            'out tests' => array('out(name,[foo,bar]', array(
```

vs. this

``` php
            'out tests' => array(
                'out(name,[foo,bar]',
```

You should really tell your IDE to not change unrelated stuff to what you're working on or start staging your changes on a line-by-line base.
It's not like I'm against this kind of cleanup, but if it is being done in an automated fashion it should be on its own branch so I don't need to go through each commit of a PR to get what really changed.
